debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/transforms/DecodeLogicalDecodingMessageContent.java

```diff
@@ -0,0 +1,198 @@
+
+        final Schema updatedValueSchema = getUpdatedValueSchema(logicalDecodingMessageContent.schema(), recordWithoutMessageField.valueSchema());
+        final Struct updatedValue = getUpdatedValue(updatedValueSchema, originalValue, logicalDecodingMessageContent);
+
+        return record.newRecord(
+                record.topic(),
+                record.kafkaPartition(),
+                null,
+                // clear prefix of a logical decoding message
+                null,
+                updatedValueSchema,
+                updatedValue,
+                record.timestamp(),
+                record.headers());
+    }
+
+    private Struct getLogicalDecodingMessageContent(Struct valueStruct) {
+        Struct logicalDecodingMessageStruct = requireStruct(valueStruct.get(DEBEZIUM_LOGICAL_DECODING_MESSAGE_KEY),
+                "Retrieve content of a logical decoding message");
+
+        if (logicalDecodingMessageStruct.schema().field(DEBEZIUM_LOGICAL_DECODING_MESSAGE_CONTENT_KEY).schema().type() != Schema.Type.BYTES) {
+            throw new DataException("The content of a logical decoding message is non-binary");
+        }
+
+        byte[] logicalDecodingMessageContentBytes = logicalDecodingMessageStruct.getBytes(DEBEZIUM_LOGICAL_DECODING_MESSAGE_CONTENT_KEY);
+        return convertLogicalDecodingMessageContentBytesToStruct(logicalDecodingMessageContentBytes);
+    }
+
+    private Struct convertLogicalDecodingMessageContentBytesToStruct(byte[] logicalDecodingMessageContent) {
+        final String logicalDecodingMessageContentString = new String(logicalDecodingMessageContent);
+        try {
+            // parse and get Jackson JsonNode
+            final JsonNode logicalDecodingMessageContentJson = parseLogicalDecodingMessageContentJsonString(logicalDecodingMessageContentString);
+            // get schema of a logical decoding message content
+            Schema logicalDecodingMessageContentSchema = jsonSchemaData.toConnectSchema(null, logicalDecodingMessageContentJson);
+            // get Struct of a logical decoding message content
+            return (Struct) jsonSchemaData.toConnectData(logicalDecodingMessageContentJson, logicalDecodingMessageContentSchema);
+        }
+        catch (Exception e) {
+            LOGGER.warn("Conversion of logical decoding message content failed", e);
+            throw new DataException("Conversion of logical decoding message content failed");
+        }
+    }
+
+    private JsonNode parseLogicalDecodingMessageContentJsonString(String logicalDecodingMessageContentJsonString) throws Exception {
+        if (logicalDecodingMessageContentJsonString.startsWith("{") || logicalDecodingMessageContentJsonString.startsWith("[")) {
+            return objectMapper.readTree(logicalDecodingMessageContentJsonString);
+        }
+        throw new Exception("Unable to parse logical decoding message content JSON string starting with '" + logicalDecodingMessageContentJsonString.charAt(0) + "'");
```

Comment:
Is it possible to use a non-generic exception? Does it need to be a checked one?